src/resources.js

```diff
@@ -512,7 +512,7 @@ export class Resources {
     const isScrollingStopped = (Math.abs(this.lastVelocity_) < 1e-2 &&
         now - this.lastScrollTime_ > MUTATE_DEFER_DELAY_ ||
         now - this.lastScrollTime_ > MUTATE_DEFER_DELAY_ * 2);
-    const offset = 10;
+    const topOffset = viewportRect.height / 10;
```

Comment:
Actually would you mind spelling out all the rules implemented here in a single comment block. I have some trouble following what is actually implemented.